jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/cloudtrail/DenyPublicAccessS3BucketStoresCloudTrailLogs.java

```diff
@@ -0,0 +1,150 @@
+                LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+                Annotation annotation = null;
+                annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                annotation.put(PacmanSdkConstants.DESCRIPTION,"Ensure S3 bucket that stores cloudtrail Logs is not publicly accessible");
+                annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                annotation.put(PacmanRuleConstants.CATEGORY, category);
+                annotation.put(PacmanRuleConstants.NAME,resourceAttributes.get(PacmanRuleConstants.NAME));
+                issue.put(PacmanRuleConstants.VIOLATION_REASON,
+                        ruleParam.get(PacmanRuleConstants.RULE_ID) + " Violation Found!");
+                issueList.add(issue);
+                annotation.put(PacmanRuleConstants.ISSUE_DETAILS, issueList.toString());
+                logger.debug("S3 bucket that stores cloudtrail Logs is publicly accessible");
+                return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                        annotation);
+            }
+        }
+        logger.debug("S3 bucket that stores cloudtrail Logs is not publicly accessible");
+        logger.debug("========DenyPublicAccessS3BucketStoresCloudTrailLogs ended=========");
+        return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+
+    private boolean checkForS3BucketCloudtrailLogsPublicAccess(String esUrl, Map<String, Object> mustFilter) throws Exception {
+        logger.info("Validating the resource data from elastic search. ES URL:{}, FilterMap : {}", esUrl, mustFilter);
+        boolean validationResult = false;
+        JsonParser parser = new JsonParser();
+        JsonObject resultJson = RulesElasticSearchRepositoryUtil.getQueryDetailsFromES(esUrl, mustFilter,
+                new HashMap<>(),
+                HashMultimap.create(), null, 0, new HashMap<>(), null, null);
+
+        if (resultJson.has(PacmanRuleConstants.HITS)) {
+            String hitsString = resultJson.get(PacmanRuleConstants.HITS).toString();
+            JsonObject hitsJson = JsonParser.parseString(hitsString).getAsJsonObject();
+            JsonArray hitsJsonArray = hitsJson.getAsJsonObject().get(PacmanRuleConstants.HITS).getAsJsonArray();
+            if (!hitsJsonArray.isEmpty()) {
+                    JsonObject source = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                            .get(PacmanRuleConstants.SOURCE);
+                    if (source.has(PacmanRuleConstants.ES_BKT_POLICY_ATTRIBUTE)) {
+                        String bucketPolicyString = source.get(PacmanRuleConstants.ES_BKT_POLICY_ATTRIBUTE).getAsString();
+
+                        JsonObject bucketPolicyObject = null;
+                        try {
+                            bucketPolicyObject = new Gson().fromJson(bucketPolicyString, JsonObject.class);
```

Comment:
what is happening here? Could you please elaborate a bit?

Reply:
As shown below , the json member "bucketpolicy" has its value in form of string. Therefore we need to get it in string then convert to json object else we get exception :  Not a JSON Object


"bucketpolicy":"{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"AWSCloudTrailAclCheck20150319\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"cloudtrail.amazonaws.com\"},\"Action\":\"s3:GetBucketAcl\",\"Resource\":\"arn:aws:s3:::aws-cloudtrail-logs-616767712067-cacc9162\",\"Condition\":{\"StringEquals\":{\"AWS:SourceArn\":\"arn:aws:cloudtrail:us-east-1:616767712067:trail/management-events\"}}},{\"Sid\":\"AWSCloudTrailWrite20150319\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"cloudtrail.amazonaws.com\"},\"Action\":\"s3:PutObject\",\"Resource\":\"arn:aws:s3:::aws-cloudtrail-logs-616767712067-cacc9162/AWSLogs/616767712067/*\",\"Condition\":{\"StringEquals\":{\"s3:x-amz-acl\":\"bucket-owner-full-control\",\"AWS:SourceArn\":\"arn:aws:cloudtrail:us-east-1:616767712067:trail/management-events\"}}}]}"